### ISSUE TITLE
ttnn.max on float32 returns a value not present in the input (global reduce, no dim)

### ISSUE THREAD

**Issue by Aswinmcw** [tenstorrent/tt-metal#51889]
### Summary

`ttnn.max` with no `dim` (global reduce) returns a value that **is not present in the input tensor** when the input is `float32`. `max` is a selection operation, so its result must be one of the input elements.

On `bfloat16` the same op returns the exact element. Only `float32` is affected.

### Minimal repro

```python
import torch, ttnn

device = ttnn.open_mesh_device(mesh_shape=ttnn.MeshShape(1, 1))
torch.manual_seed(0)
src = torch.randn(1, 1, 64, 64, dtype=torch.float32)
tt = ttnn.from_torch(src, dtype=ttnn.float32, layout=ttnn.TILE_LAYOUT, device=device)

got = ttnn.to_torch(ttnn.max(tt)).flatten()[0]      # global reduce: no `dim`
print("torch.max        :", torch.max(src).item())
print("ttnn.max         :", got.item())
print("present in input?", bool((src.flatten() == got).any()))
ttnn.close_mesh_device(device)
```

### Expected

```
present in input? True
```

### Actual

```
torch.max        : 4.1014933586120605
ttnn.max         : 4.09814453125
present in input? False
```

### Scope

6/6 float32 cases return a value absent from the input; 6/6 bfloat16 cases return the exact element.

| dtype | shape | seed | `torch.max` | `ttnn.max` | in input? | rel err |
|---|---|---|---|---|---|---|
| bfloat16 | (16, 2, 32, 3) | 0 | 4.093750000 | 4.093750000 | **True** | 0 |
| bfloat16 | (16, 2, 32, 24) | 1 | 3.796875000 | 3.796875000 | **True** | 0 |
| bfloat16 | (1, 1, 64, 64) | 0 | 4.093750000 | 4.093750000 | **True** | 0 |
| float32 | (16, 2, 32, 3) | 0 | 4.101493359 | 4.098144531 | **False** | 8.16e-04 |
| float32 | (16, 2, 32, 3) | 1 | 3.362807751 | 3.361572266 | **False** | 3.67e-04 |
| float32 | (16, 2, 32, 24) | 1 | 3.802687168 | 3.801269531 | **False** | 3.73e-04 |
| float32 | (1, 1, 64, 64) | 1 | 3.405680895 | 3.406005859 | **False** | 9.54e-05 |

The error magnitude (1e-4 … 8e-4) is consistent with bfloat16 precision, which suggests the reduce runs at reduced internal precision and returns the rounded value rather than the selected element.

A fuller script covering both dtypes and several shapes is at `repro_max_float32_selection.py` (exits 1 when reproduced).

### Why this isn't caught today

`tests/ttnn/unit_tests/operations/reduce/test_max.py::test_max_global` is the only test covering the no-`dim` path, and it uses **bfloat16** input with **`assert_equal`**:

```python
torch_input_tensor = torch_random((batch_size, h, w), -100, 100, dtype=torch.bfloat16)
torch_output_tensor = torch.max(torch_input_tensor)
output_tensor = ttnn.max(input_tensor)
assert_equal(torch_output_tensor, output_tensor)
```

That combination passes trivially, so the float32 path is never exercised.

### How it surfaced

Found in the model-traced sweeps, where FLOAT32 `max` vectors with `dim` absent reported `PCC 0.0` (shapes `(16,2,32,{3,6,12,24})`, e.g. vector `035bf99f081b`). PCC is undefined for a scalar result, so the comparison falls back to `torch.allclose`, and the discrepancy above exceeds its tolerance.

### Question for op owners

Is reduced-precision internal accumulation expected for `float32` reductions? If so, callers need to know `max` may return a value not in the input, which is surprising for a selection op and means exact-equality assertions are invalid on `float32`. If not, this looks like a genuine precision bug in the reduce path.

### Environment

- commit `97897756cde`
- `wormhole_b0`, N300 (2 chips), single-device mesh (1, 1)
- firmware bundle 19.7.1, KMD 2.5.0
- torch 2.11.0+cpu

**Comment by bbradelTT** [comment]
@iwroszTT please check, or assign someone from ign to check, if this path is going through sfpu, which might not be affected by the fpu bug.
